utils/src/main/scala/za/co/absa/enceladus/utils/general/ProjectMetadata.scala

```diff
@@ -15,8 +15,10 @@
 
 package za.co.absa.enceladus.utils.general
 
-object ProjectMetadata {
-
-  val enceladusVersion: String = getClass.getPackage.getImplementationVersion
+trait ProjectMetadata {
 
+  def enceladusVersion(): String = {
+    getClass.getPackage.getImplementationVersion
+  }
 }
+object ProjectMetadata extends ProjectMetadata
```

Comment:
Why implement a selfless trait pattern? If this is not used as a mixin it will output version of the utils module, will it not?